web_search_with_and/static/src/js/search.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('web_search_with_and', function (require) {
+    "use strict";
+
+    var SearchView = require('web.SearchView');
+    var Backbone = window.Backbone;
+
+    SearchView.include({
+        // Override the base method to detect a 'shift' event
+        select_completion: function (e, ui) {
+            if (e.shiftKey
+                && ui.item.facet.values
+                && ui.item.facet.values.length
+                && String(ui.item.facet.values[0].value).trim() !== "") {
+                // In case of an 'AND' search a new facet is added regarding of the previous facets
+                e.preventDefault();
+
+                this.query.add(ui.item.facet, {shiftKey: true});
+            } else {
+
+                return this._super.apply(this, arguments);
+            }
+        }
+    });
+
+    SearchView.SearchQuery.prototype = SearchView.SearchQuery.extend({
```

Comment:
To be honest, before using prototype, I tried 

1. `SearchView.SearchQuery.include({}); `
2. `SearchView.SearchQuery = SearchView.SearchQuery.extend({});`

But it doesn't work, for I need to place the code (in the inheritance stack) before the extend in odoo/web/static/src/js/views/search_view.js, extract : 

```
var SearchQuery = Backbone.Collection.extend({
    ...
    add: function (values, options) {
        ...
        _(values).each(function (value) {
            ...
            Backbone.Collection.prototype.add.call(this, model, options);

...
_.extend(SearchView, {
    SearchQuery: SearchQuery,
    Facet: Facet,
});

return SearchView;
```
Not sure if I make sense here, but it was crystal clear when I implemented it